### DISCUSSION TITLE
LuceneImpactSearcher with UniCOIL Model

### DISCUSSION THREAD

**Discussion by hieudx149** [castorini/pyserini#1264] [Q&A] [answered]
Hi there, I follow this repo (https://github.com/luyug/COIL) to train a UniCOIL model with my own dataset (in Vietnamese), now I want to create a search API base on this UniCOIL model, follow this code (https://github.com/castorini/pyserini/blob/master/pyserini/search/lucene/__main__.py)  and this script (https://github.com/castorini/pyserini/blob/master/docs/experiments-unicoil.md) I have to use LuceneImpactSearcher instead LuceneSearcher, but LuceneImpactSearcher need query_encoder argument which use to encode query for searching, but I use another encoder to encode query first then do searcher.search(text) with text was encoded by the encoder, is there any way to ignore query_encoder argument when using LuceneImpactSearcher

**Comment by lintool** [comment, marked as answer]
Yes, see the `--encoded-queries` options here as an example: https://github.com/castorini/pyserini/blob/master/docs/experiments-tct_colbert-v2.md

**Reply by hieudx149** [reply]
Thanks for your response, I solved this
